view/infinite.go

```diff
@@ -0,0 +1,431 @@
+package view
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/jsdelivr/globalping-cli/client"
+	"github.com/jsdelivr/globalping-cli/model"
+	"github.com/mattn/go-runewidth"
+	"github.com/pterm/pterm"
+)
+
+// Table defaults
+var (
+	colSeparator = " | "
+)
+
+func OutputInfinite(id string, ctx *model.Context) error {
+	fetcher := client.NewMeasurementsFetcher(client.ApiUrl)
+	res, err := fetcher.GetMeasurement(id)
+	if err != nil {
+		return err
+	}
+	// Probe may not have started yet
+	for len(res.Results) == 0 {
+		time.Sleep(ctx.APIMinInterval)
+		res, err = fetcher.GetMeasurement(id)
+		if err != nil {
+			return err
+		}
+	}
+
+	if ctx.Latency || ctx.JsonOutput {
+		for res.Status == model.StatusInProgress {
+			time.Sleep(ctx.APIMinInterval)
+			res, err = fetcher.GetMeasurement(res.ID)
+			if err != nil {
+				return err
+			}
+		}
+		if ctx.Latency {
+			return OutputLatency(id, res, *ctx)
+		}
+		if ctx.JsonOutput {
+			return OutputJson(id, fetcher, *ctx)
+		}
+	}
+
+	if len(res.Results) == 1 {
+		return outputSingleLocation(fetcher, res, ctx)
+	}
+	return outputMultipleLocations(fetcher, res, ctx)
+}
+
+func outputSingleLocation(
+	fetcher client.MeasurementsFetcher,
+	res *model.GetMeasurement,
+	ctx *model.Context,
+) error {
+	if len(ctx.Stats) == 0 {
+		ctx.Stats = make([]model.MeasurementStats, 1)
+	}
+	printHeader := true
+	linesPrinted := 0
+	var err error
+	for {
+		measurement := &res.Results[0]
+		if measurement.Result.RawOutput != "" {
+			parsedOutput, err := parsePingRawOutput(measurement, ctx.Stats[0].Sent)
+			if err != nil {
+				return err
+			}
+			if printHeader && ctx.Stats[0].Sent == 0 {
+				fmt.Println(generateProbeInfo(measurement, !ctx.CI))
+				fmt.Printf("PING %s (%s) 56(84) bytes of data.\n",
+					measurement.Result.ResolvedHostname,
+					measurement.Result.ResolvedAddress,
+				)
+				printHeader = false
+			}
+			for linesPrinted < len(parsedOutput.RawPacketLines) {
+				fmt.Println(parsedOutput.RawPacketLines[linesPrinted])
+				linesPrinted++
+			}
+			if res.Status != model.StatusInProgress {
+				ctx.Stats[0].Sent += parsedOutput.Stats.Total
+			}
+		}
+		if res.Status != model.StatusInProgress {
+			break
+		}
+		time.Sleep(ctx.APIMinInterval)
+		res, err = fetcher.GetMeasurement(res.ID)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func outputMultipleLocations(
```

Comment:
There seems to be a bug (happens not always but often) - when the first measurement ends, the number of packets jumps from 16 to 32 as if it was counted twice:
![Animation2](https://github.com/jsdelivr/globalping-cli/assets/6192491/98a756d7-c503-41c9-b7bc-a17428a0e56e)